### DISCUSSION TITLE
Keeping relay on until alarm value is reached

### DISCUSSION THREAD

**Discussion by threedom** [theyosh/TerrariumPI#714] [Q&A] [answered]
I'm trying to configure a heat lamp in a way that if the ambient temperature exceeds a certain value (i.e. if it is a very warm day and the terrarium is warm enough without the heat lamp), such that the lamp switches off. Otherwise I want it to always be on. 

I cannot figure out what settings in the Area controls would enable that. Whatever value I enter for power on / settle time for either low or high alarm don't seem to do the trick. Instead after the specified time (e.g. 30s power on) the relay just switches off (the lamp switches off) and in the next cycle it switches on again. 

Since I don't want my critter to have an on /off heat lamp all day, I'd appreciate any advice!

**Comment by theyosh** [comment, marked as answer]
Hi, yes this is possible using the 'Ignore high/low alarm' toggle in front of the relays. More information is here: https://github.com/theyosh/TerrariumPI/issues/669#issuecomment-1072218476

So, you create a heating area with a mode that fits. Lets say timer. Then you have to specify at least the begin and end time where the relay(s) should be on. Or use weather, to use sun rise and sun set values.
Then the other fields with duration, power on time and settle time should contain a `0`. So this will not affect the logic, and therefore the relays will be on during the hole time between begin and end time.
Next you select the temperature sensor that should control the heater.
And final, toggle the `Ignore low alarm` to on, so that the relays will stay on, until max heat is reached. 

But if the temperature is around the max temperature, you will still get the toggle on/off effect in order to keep the temperature at the max value.

With the settle time, you can specify the time between two toggle on actions. So, here you can set a value of 300 seconds which means that when the heat lamp toggles off, it takes 5 minutes before it will be toggled on. This way you can reduce the amount of on and off toggle actions.

**Reply by threedom** [reply]
So we're having a few warm days in a row and now I see that it doesn't seem to be working as intended:

Having all settings as described by you (see screenshots), the heatlamp doesn't ever switch off when the temperature exceeds the alarm value. When I add an "off duration" it does switch off once the temperature is too high, however, when the temperature is below the alarm, it switches on and off in intervals (every 30s or so). 

If I want the lamp to be on all the time when the temperature is too low and off all the time when it is too high (with the downside that it may switch on and off while the temperature is close to the alarm value), in what field would I have to enter what kind of value? I tried values for the "off duration" field but as I said, that leads to rhythmical switching on and off even when the value is well below the alarm value (and does not exceed the alarm value ever).

I should say that my temp probe is outside of the direct influence of the heatlamp. It is measuring ambient temperature and I am just trying to switch off the lamp when it is so warm that it would be unnecessary to "pump" further heat into the vivarium.

![image](https://user-images.githubusercontent.com/11461427/188760957-29dbe626-9e3f-42d5-9d05-a480b9ab5a4e.png)


![image](https://user-images.githubusercontent.com/11461427/188760984-5329e93e-e1d7-44ca-9291-f3bf6f07654f.png)

**Reply by theyosh** [reply]
Hmm, my first idea is that you should remove the relay from the high alarm. A relay should be used only once. Now you are using it twice, and this could give strange effects.

The high alarm part should only used for cooling hardware. And you lamp is a heating lamp. So only use that on the low alarm part. I think that should do it. All other fields have the correct values.

**Reply by threedom** [reply]
That solved it, thanks a lot! 
It might make sense adding the info that only high or low alarm should be used to the setup instructions. Because intuitively I'd expect to have to fill in both. (And I actually just found another issue item where someone had this problem a while ago).
